marconi-sidechain/doc/adr/01-sdd-computation.adoc

```diff
@@ -0,0 +1,203 @@
+= ADR 1: Stake pool delegation distribution computation
+
+Date: 2023-05-01
+
+== Author(s)
+
+@koslambrou <konstantinos.lambrou@iohk.io>
+
+== Status
+
+Draft
+
+== Context
+
+The Sidechain application needs to know the Cardano blockchain active stake pool delegation distribution (SDD) for epoch `n`.
+It will use that information to register the committee that will be used to validate sidechain blocks in epoch `n+1`.
+
+Here's the list of assumptions:
+
+* `marconi-sidechain` will be run by a Stake Pool Operator (SPO)
+* `marconi-sidechain` will communicate with a relay node (not the block producing node)
+
+The main architectural goal we are trying to achieve is to reduce as much as possible the hardware resource usage of a SPO.
+Therefore, indexing the SDD per epoch should be as resource efficient as possible.
+
+== Decision(s)
+
+TBD
+
+== Possible solutions
+
+We will describe all the potential and discussed solutions to index the SDD of each epoch in the following sections.
+
+As a summary, we provide the following caracteristics table and indicate how each solution responds to them.
+This assessment is measured in regards to two components: Cardano Node and Marconi.
+
+[cols="1,1,1,1,1"]
+|===
+|Caracteristics | Solution 1 | Solution 2 | Solution 3 | Solution 4
+
+|Number of LedgerState computations | 2 | 1 | 1 | 1
+|Initial node resync required | No | Yes | Yes | Yes
+|Required node resync after Marconi logic change | No | No | Yes | Yes
```

Comment:
After reading the definition of solution 3 and 4 I don't get why we need a resync after logic change if it doesn't impact the SDD computation. Is it because we can't define, in the general case whether or not we will need to go through the SDD computation again to support the new logic?

Reply:
If you change the indexing logic of `marconi-sidechain`, then you need to resync it from scratch, right? If that's the case, then you _also_ need to resync `cardano-node` from scratch for solutions 3 and 4.

Reply:
We may find a way to resync only the parts that are impacted by a change of logic (especially with solution 4): so yes, we need to resync from scratch, but if we don't change the SDD computation we might keep our SDD indexing and have a lighter resync.
 It would be moving the reindexing problem we have in Marconi to a resync problem in a Marconi relay node, and thus limit the negative impact.

Reply:
Oh but yes, in the case of side-chain, it would still impact the relay node, even if it's faster. I get the point now.